File: Analise/generate_docx.js
const fs = require('fs');
const { Document, Packer, Paragraph, HeadingLevel, TextRun } = require('docx');

const mdPath = 'Analise/relatorio-executivo.md';
const outPath = 'Analise/RELATORIO_EXECUTIVO_FINAL_CODEBASE_FCDDDPATTERNS.docx';

function lineToParagraph(line) {
  if (line.startsWith('# ')) {
    return new Paragraph({ text: line.slice(2).trim(), heading: HeadingLevel.TITLE });
  }
  if (line.startsWith('## ')) {
    return new Paragraph({ text: line.slice(3).trim(), heading: HeadingLevel.HEADING_1 });
  }
  if (line.startsWith('### ')) {
    return new Paragraph({ text: line.slice(4).trim(), heading: HeadingLevel.HEADING_2 });
  }
  if (line.startsWith('- ')) {
    return new Paragraph({ text: line.slice(2).trim(), bullet: { level: 0 } });
  }
  if (line.trim() === '---') {
    // page break
    return new Paragraph({ children: [new TextRun({ text: '', break: 1 })] });
  }
  return new Paragraph({ text: line });
}

function main() {
  const md = fs.readFileSync(mdPath, 'utf8');
  const lines = md.split(/\r?\n/);
  const body = [];
  for (const line of lines) {
    if (line.trim() === '') {
      body.push(new Paragraph(''));
    } else {
      body.push(lineToParagraph(line));
    }
  }
  const doc = new Document({ creator: 'Analise', title: 'Relatório Executivo', description: 'Relatório Executivo — Análise do codebase', sections: [{ children: body }] });
  Packer.toBuffer(doc).then((buffer) => {
    fs.writeFileSync(outPath, buffer);
    console.log('Saved', outPath);
  }).catch((err) => {
    console.error(err);
  });
}

main();

File: Analise/generate_docx.py
from docx import Document
import sys

md_path = 'Analise/relatorio-executivo.md'
docx_path = 'Analise/relatorio-executivo.docx'

def add_paragraph_from_line(doc, line):
    if line.startswith('# '):
        doc.add_heading(line[2:].strip(), level=1)
    elif line.startswith('## '):
        doc.add_heading(line[3:].strip(), level=2)
    elif line.startswith('### '):
        doc.add_heading(line[4:].strip(), level=3)
    elif line.startswith('- '):
        doc.add_paragraph(line[2:].strip(), style='List Bullet')
    elif line.strip() == '---':
        doc.add_page_break()
    else:
        doc.add_paragraph(line)


def main():
    doc = Document()
    with open(md_path, 'r', encoding='utf-8') as f:
        for raw in f:
            add_paragraph_from_line(doc, raw.rstrip('\n'))
    doc.save(docx_path)
    print('Saved', docx_path)

if __name__ == '__main__':
    main()
